### PR TITLE
Sync Reduction builder after upstreaming

### PR DESCRIPTION
**Pull request by clementval**
Sync after https://reviews.llvm.org/D114460

### CODE REVIEW

File: flang/include/flang/Optimizer/Builder/Runtime/Reduction.h [hunk 1]
```diff
@@ -18,33 +25,33 @@ class FirOpBuilder;
...
 
-/// Generate call to any runtime routine. This version of any is specialized
+/// Generate call to `any` runtime routine. This version of `any` is specialized
 /// for rank 1 mask arguments.
 /// This calls the version that returns a scalar logical value.
 mlir::Value genAny(fir::FirOpBuilder &builder, mlir::Location loc,
                    mlir::Value maskBox, mlir::Value dim);
 
-/// Generate call to Count runtime routine. This routine is a specialized
+/// Generate call to `count` runtime routine. This routine is a specialized
```
Comment (schweitzpgi) [inline flang/include/flang/Optimizer/Builder/Runtime/Reduction.h:54]: These comment changes are strange. The runtime routine is, in fact, named "Count" (with a name mangling macro wrapper). Generally, flang uses all caps names to refer to Fortran artifacts, so "COUNT" would be the typical way of referring to the intrinsic in, say, a presentation slide.

It's arguable that `count` isn't just wrong even. The backquotes suggest it is an identifier in the code, such as an argument name. It clearly is not.

Comment (clementval) [reply flang/include/flang/Optimizer/Builder/Runtime/Reduction.h:54]: Well it was ask be reviewers. I'm personally fine reverting to the original routine name.

File: flang/include/flang/Optimizer/Builder/Runtime/Reduction.h [hunk 2]
```diff
@@ -56,7 +63,7 @@ void genCountDim(fir::FirOpBuilder &builder, mlir::Location loc,
                  mlir::Value resultBox, mlir::Value maskBox, mlir::Value dim,
                  mlir::Value kind);
 
-/// Generate call to DotProduct intrinsic runtime routine.
+/// Generate call to `dot_product` intrinsic runtime routine.
```
Comment (schweitzpgi) [inline flang/include/flang/Optimizer/Builder/Runtime/Reduction.h:66]:
```suggestion
/// Generate call to `DotProduct` intrinsic runtime routine.
```

File: flang/lib/Optimizer/Builder/Runtime/Reduction.cpp [hunk 3]
```diff
@@ -451,8 +451,8 @@ genReduction5Args(FN func, fir::FirOpBuilder &builder, mlir::Location loc,
   builder.create<fir::CallOp>(loc, func, args);
 }
 
-/// Generate call to all runtime routine.
-/// This calls the descriptor based runtime call implementation of the all
+/// Generate call to `all` runtime routine.
```
Comment (schweitzpgi) [inline flang/lib/Optimizer/Builder/Runtime/Reduction.cpp:454]:
```suggestion
/// Generate call to `All` runtime routine.
```

File: flang/lib/Optimizer/Builder/Runtime/Reduction.cpp [hunk 4]
```diff
@@ -461,8 +461,8 @@ void fir::runtime::genAllDescriptor(fir::FirOpBuilder &builder,
   genReduction2Args(allFunc, builder, loc, resultBox, maskBox, dim);
 }
 
-/// Generate call to any runtime routine.
-/// This calls the descriptor based runtime call implementation of the any
+/// Generate call to `any` runtime routine.
```
Comment (schweitzpgi) [inline flang/lib/Optimizer/Builder/Runtime/Reduction.cpp:464]:
```suggestion
/// Generate call to `Any` runtime routine.
```

File: flang/lib/Optimizer/Builder/Runtime/Reduction.cpp [hunk 5]
```diff
@@ -471,23 +471,23 @@ void fir::runtime::genAnyDescriptor(fir::FirOpBuilder &builder,
   genReduction2Args(anyFunc, builder, loc, resultBox, maskBox, dim);
 }
 
-/// Generate call to All intrinsic runtime routine. This routine is
+/// Generate call to `all` intrinsic runtime routine. This routine is
```
Comment (schweitzpgi) [inline flang/lib/Optimizer/Builder/Runtime/Reduction.cpp:474]:
```suggestion
/// Generate call to `All` intrinsic runtime routine. This routine is
```

File: flang/lib/Optimizer/Builder/Runtime/Reduction.cpp [hunk 6]
```diff
@@ -471,23 +471,23 @@ void fir::runtime::genAnyDescriptor(fir::FirOpBuilder &builder,
...
 /// specialized for mask arguments with rank == 1.
 mlir::Value fir::runtime::genAll(fir::FirOpBuilder &builder, mlir::Location loc,
                                  mlir::Value maskBox, mlir::Value dim) {
   auto allFunc = fir::runtime::getRuntimeFunc<mkRTKey(All)>(loc, builder);
   return genSpecial2Args(allFunc, builder, loc, maskBox, dim);
 }
 
-/// Generate call to Any intrinsic runtime routine. This routine is
+/// Generate call to `any` intrinsic runtime routine. This routine is
```
Comment (schweitzpgi) [inline flang/lib/Optimizer/Builder/Runtime/Reduction.cpp:482]:
```suggestion
/// Generate call to `Any` intrinsic runtime routine. This routine is
```

File: flang/lib/Optimizer/Builder/Runtime/Reduction.cpp [hunk 7]
```diff
@@ -471,23 +471,23 @@ void fir::runtime::genAnyDescriptor(fir::FirOpBuilder &builder,
...
 /// specialized for mask arguments with rank == 1.
 mlir::Value fir::runtime::genAny(fir::FirOpBuilder &builder, mlir::Location loc,
                                  mlir::Value maskBox, mlir::Value dim) {
   auto anyFunc = fir::runtime::getRuntimeFunc<mkRTKey(Any)>(loc, builder);
   return genSpecial2Args(anyFunc, builder, loc, maskBox, dim);
 }
 
-/// Generate call to Count runtime routine. This routine is a specialized
+/// Generate call to `count` runtime routine. This routine is a specialized
```
Comment (schweitzpgi) [inline flang/lib/Optimizer/Builder/Runtime/Reduction.cpp:490]:
```suggestion
/// Generate call to `Count` runtime routine. This routine is a specialized
```

File: flang/lib/Optimizer/Builder/Runtime/Reduction.cpp [hunk 8]
```diff
@@ -512,7 +512,7 @@ void fir::runtime::genCountDim(fir::FirOpBuilder &builder, mlir::Location loc,
   builder.create<fir::CallOp>(loc, func, args);
 }
 
-/// Generate call to Maxloc intrinsic runtime routine. This is the version
+/// Generate call to `maxloc` intrinsic runtime routine. This is the version
```
Comment (schweitzpgi) [inline flang/lib/Optimizer/Builder/Runtime/Reduction.cpp:515]:
```suggestion
/// Generate call to `Maxloc` intrinsic runtime routine. This is the version
```

File: flang/lib/Optimizer/Builder/Runtime/Reduction.cpp [hunk 9]
```diff
@@ -523,7 +523,7 @@ void fir::runtime::genMaxloc(fir::FirOpBuilder &builder, mlir::Location loc,
                     back);
 }
 
-/// Generate call to Maxloc intrinsic runtime routine. This is the version
+/// Generate call to `maxloc` intrinsic runtime routine. This is the version
```
Comment (schweitzpgi) [inline flang/lib/Optimizer/Builder/Runtime/Reduction.cpp:526]:
```suggestion
/// Generate call to `Maxloc` intrinsic runtime routine. This is the version
```

File: flang/lib/Optimizer/Builder/Runtime/Reduction.cpp [hunk 10]
```diff
@@ -534,7 +534,7 @@ void fir::runtime::genMaxlocDim(fir::FirOpBuilder &builder, mlir::Location loc,
                     back);
 }
 
-/// Generate call to Maxval intrinsic runtime routine. This is the version
+/// Generate call to `maxval` intrinsic runtime routine. This is the version
```
Comment (schweitzpgi) [inline flang/lib/Optimizer/Builder/Runtime/Reduction.cpp:537]:
```suggestion
/// Generate call to `Maxval` intrinsic runtime routine. This is the version
```

File: flang/lib/Optimizer/Builder/Runtime/Reduction.cpp [hunk 11]
```diff
@@ -581,7 +581,7 @@ mlir::Value fir::runtime::genMaxval(fir::FirOpBuilder &builder,
   return builder.create<fir::CallOp>(loc, func, args).getResult(0);
 }
 
-/// Generate call to Maxval intrinsic runtime routine. This is the version
+/// Generate call to `maxval` intrinsic runtime routine. This is the version
```
Comment (schweitzpgi) [inline flang/lib/Optimizer/Builder/Runtime/Reduction.cpp:584]:
```suggestion
/// Generate call to `Maxval` intrinsic runtime routine. This is the version
```

File: flang/lib/Optimizer/Builder/Runtime/Reduction.cpp [hunk 12]
```diff
@@ -590,7 +590,7 @@ void fir::runtime::genMaxvalDim(fir::FirOpBuilder &builder, mlir::Location loc,
   genReduction3Args(func, builder, loc, resultBox, arrayBox, dim, maskBox);
 }
 
-/// Generate call to Maxval intrinsic runtime routine. This is the version
+/// Generate call to `maxval` intrinsic runtime routine. This is the version
```
Comment (schweitzpgi) [inline flang/lib/Optimizer/Builder/Runtime/Reduction.cpp:593]:
```suggestion
/// Generate call to `Maxval` intrinsic runtime routine. This is the version
```

File: flang/lib/Optimizer/Builder/Runtime/Reduction.cpp [hunk 13]
```diff
@@ -606,7 +606,7 @@ void fir::runtime::genMaxvalChar(fir::FirOpBuilder &builder, mlir::Location loc,
   builder.create<fir::CallOp>(loc, func, args);
 }
 
-/// Generate call to Minloc intrinsic runtime routine. This is the version
+/// Generate call to `minloc` intrinsic runtime routine. This is the version
```
Comment (schweitzpgi) [inline flang/lib/Optimizer/Builder/Runtime/Reduction.cpp:609]:
```suggestion
/// Generate call to `Minloc` intrinsic runtime routine. This is the version
```

File: flang/lib/Optimizer/Builder/Runtime/Reduction.cpp [hunk 14]
```diff
@@ -617,7 +617,7 @@ void fir::runtime::genMinloc(fir::FirOpBuilder &builder, mlir::Location loc,
                     back);
 }
 
-/// Generate call to Minloc intrinsic runtime routine. This is the version
+/// Generate call to `minloc` intrinsic runtime routine. This is the version
```
Comment (schweitzpgi) [inline flang/lib/Optimizer/Builder/Runtime/Reduction.cpp:620]:
```suggestion
/// Generate call to `Minloc` intrinsic runtime routine. This is the version
```

File: flang/lib/Optimizer/Builder/Runtime/Reduction.cpp [hunk 15]
```diff
@@ -628,7 +628,7 @@ void fir::runtime::genMinlocDim(fir::FirOpBuilder &builder, mlir::Location loc,
                     back);
 }
 
-/// Generate call to Minval intrinsic runtime routine. This is the version
+/// Generate call to `minval` intrinsic runtime routine. This is the version
```
Comment (schweitzpgi) [inline flang/lib/Optimizer/Builder/Runtime/Reduction.cpp:631]:
```suggestion
/// Generate call to `Minval` intrinsic runtime routine. This is the version
```

File: flang/lib/Optimizer/Builder/Runtime/Reduction.cpp [hunk 16]
```diff
@@ -637,7 +637,7 @@ void fir::runtime::genMinvalDim(fir::FirOpBuilder &builder, mlir::Location loc,
   genReduction3Args(func, builder, loc, resultBox, arrayBox, dim, maskBox);
 }
 
-/// Generate call to Minval intrinsic runtime routine. This is the version
+/// Generate call to `minval` intrinsic runtime routine. This is the version
```
Comment (schweitzpgi) [inline flang/lib/Optimizer/Builder/Runtime/Reduction.cpp:640]:
```suggestion
/// Generate call to `Minval` intrinsic runtime routine. This is the version
```

File: flang/lib/Optimizer/Builder/Runtime/Reduction.cpp [hunk 17]
```diff
@@ -653,7 +653,7 @@ void fir::runtime::genMinvalChar(fir::FirOpBuilder &builder, mlir::Location loc,
   builder.create<fir::CallOp>(loc, func, args);
 }
 
-/// Generate call to Minval intrinsic runtime routine. This is the version
+/// Generate call to `minval` intrinsic runtime routine. This is the version
```
Comment (schweitzpgi) [inline flang/lib/Optimizer/Builder/Runtime/Reduction.cpp:656]:
```suggestion
/// Generate call to `Minval` intrinsic runtime routine. This is the version
```

File: flang/lib/Optimizer/Builder/Runtime/Reduction.cpp [hunk 18]
```diff
@@ -700,7 +700,7 @@ mlir::Value fir::runtime::genMinval(fir::FirOpBuilder &builder,
   return builder.create<fir::CallOp>(loc, func, args).getResult(0);
 }
 
-/// Generate call to Product intrinsic runtime routine. This is the version
+/// Generate call to `product` intrinsic runtime routine. This is the version
```
Comment (schweitzpgi) [inline flang/lib/Optimizer/Builder/Runtime/Reduction.cpp:703]:
```suggestion
/// Generate call to `Product` intrinsic runtime routine. This is the version
```

File: flang/lib/Optimizer/Builder/Runtime/Reduction.cpp [hunk 19]
```diff
@@ -709,7 +709,7 @@ void fir::runtime::genProductDim(fir::FirOpBuilder &builder, mlir::Location loc,
   genReduction3Args(func, builder, loc, resultBox, arrayBox, dim, maskBox);
 }
 
-/// Generate call to Product intrinsic runtime routine. This is the version
+/// Generate call to `product` intrinsic runtime routine. This is the version
```
Comment (schweitzpgi) [inline flang/lib/Optimizer/Builder/Runtime/Reduction.cpp:712]:
```suggestion
/// Generate call to `Product` intrinsic runtime routine. This is the version
```

File: flang/lib/Optimizer/Builder/Runtime/Reduction.cpp [hunk 20]
```diff
@@ -777,6 +777,7 @@ mlir::Value fir::runtime::genProduct(fir::FirOpBuilder &builder,
   return builder.create<fir::CallOp>(loc, func, args).getResult(0);
 }
 
+/// Generate call to `dot_product` intrinsic runtime routine.
```
Comment (schweitzpgi) [inline flang/lib/Optimizer/Builder/Runtime/Reduction.cpp:780]:
```suggestion
/// Generate call to `DotProduct` intrinsic runtime routine.
```

File: flang/lib/Optimizer/Builder/Runtime/Reduction.cpp [hunk 21]
```diff
@@ -861,7 +862,7 @@ void fir::runtime::genSumDim(fir::FirOpBuilder &builder, mlir::Location loc,
   genReduction3Args(func, builder, loc, resultBox, arrayBox, dim, maskBox);
 }
 
-/// Generate call to Sum intrinsic runtime routine. This is the version
+/// Generate call to `sum` intrinsic runtime routine. This is the version
```
Comment (schweitzpgi) [inline flang/lib/Optimizer/Builder/Runtime/Reduction.cpp:865]:
```suggestion
/// Generate call to `Sum` intrinsic runtime routine. This is the version
```

**Review by schweitzpgi — APPROVED** [review]

**Review by schweitzpgi — APPROVED** [review]
+1

**Pull request merged**